integration-test-groups/foundation/scheduler/src/main/java/org/apache/camel/quarkus/component/scheduler/it/GreedyCounter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.quarkus.component.scheduler.it;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Named;
+
+@ApplicationScoped
+@Named("greedyCounter")
+public class GreedyCounter {
+    private int count = 0;
+
+    public int getCount() {
+        return count;
+    }
+
+    public void increment() {
+        count++;
+    }
```

Comment:
Note that getCount and increment() can be called from differen threads concurrently. While I do not think it matters much in how we call them from tests currently, it is still potentially dangerous (end users may copy this and use differently, we may extend the test in the future). So producing a named AtomicInteger is perhaps a better choice.

Reply:
And maybe the proposal is even to produce one `AtomicInteger` named `greedyCounter` and a second one named `mycounter`